backend/src/main/kotlin/com/hiper2d/handler/AuthHandler.kt

```diff
@@ -0,0 +1,66 @@
+package com.hiper2d.handler
+
+import com.hiper2d.auth.JwtConfigService
+import com.hiper2d.auth.model.JwtAuthenticationRequest
+import com.hiper2d.auth.model.JwtAuthenticationResponse
+import com.hiper2d.model.User
+import com.hiper2d.repository.UserRepository
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.MediaType
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.server.ServerRequest
+import org.springframework.web.reactive.function.server.ServerResponse
+import reactor.core.publisher.Mono
+
+@Component
+class AuthHandler @Autowired constructor(
+    private val userRepository: UserRepository,
+    private val jwtConfig: JwtConfigService,
+    private val encoder: PasswordEncoder
+) {
+
+    fun register(request: ServerRequest): Mono<ServerResponse> {
+        return request.bodyToMono(JwtAuthenticationRequest::class.java)
+            .flatMap { createUser(it) }
+            .flatMap {
+                ServerResponse.ok()
+                    .contentType(MediaType.APPLICATION_JSON_UTF8)
+                    .build()
+            }
+            .onErrorResume {
+                ServerResponse.status(500).build()
+            }
+    }
+
+    fun authenticate(request: ServerRequest): Mono<ServerResponse> {
+        val user = request.bodyToMono(JwtAuthenticationRequest::class.java)
+            .flatMap { findUser(it) }
+            .map { JwtAuthenticationResponse(it.username, generateJwtToken(it)) }
+
+        return ServerResponse.ok()
+            .contentType(MediaType.APPLICATION_JSON_UTF8)
+            .body(user, JwtAuthenticationResponse::class.java)
+    }
+
+    private fun createUser(token: JwtAuthenticationRequest): Mono<User> =
+        Mono.just(User(name = token.username, password = encoder.encode(token.password), roles = emptyList()))
```

Comment:
why not just 
userRepository.save(User(name = token.username, password = encoder.encode(token.password), roles = emptyList())) ?